### PR TITLE
chore:migrate ResultSet from class components to functional components

### PR DESCRIPTION
**Pull request by The-hyphen-user**
### SUMMARY

changed class components to functional

### TESTING INSTRUCTIONS

testing file still needs to be updated to test use-effect instead of component did mount/ component will receive props

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by EugeneTorap** [comment]
@The-hyphen-user Hi, we already have the same #18593 PR
You can help me to finish my #18729 #18593 #18148 PRs

File: superset-frontend/src/SqlLab/components/ResultSet/index.tsx [hunk 1]
```diff
@@ -171,106 +158,135 @@ const updateDataset = async (
...
-    // when new results comes in, save them locally and clear in store
+const ResultSet = ({
+  actions,
+  cache,
+  csv,
+  database,
+  displayLimit,
+  height,
+  query,
+  search,
```
Comment (AAfghahi) [inline superset-frontend/src/SqlLab/components/ResultSet/index.tsx:169]: it seems like this has a default value, which you should put in. Same with database, csv, showSql, and visualize.

File: superset-frontend/src/SqlLab/components/ResultSet/index.tsx [hunk 2]
```diff
@@ -664,180 +599,177 @@ export default class ResultSet extends React.PureComponent<
...
+      <div>{query.progress === 100 && <Loading position="normal" />}</div>
+      <QueryStateLabel query={query} />
+      <div>{progressMsg && <Alert type="success" message={progressMsg} />}</div>
+      <div>{query.progress !== 100 && progressBar}</div>
+      <div>{trackingUrl}</div>
+    </div>
+  );
+};
+
+ResultSet.defaultProps = {
```
Comment (AAfghahi) [inline superset-frontend/src/SqlLab/components/ResultSet/index.tsx:766]: I think you can move this in favor of stating defaults in the functional component itself.

**Review by AAfghahi — CHANGES_REQUESTED** [review]

**Comment by kgabryje** [comment]
Superseded by #21186.

**Pull request closed without merging**